### PR TITLE
backports to 1.1 2018-08-28

### PR DESCRIPTION
**Pull request by nebril**
#5300
 #5284
 #5252
 #5250
 #5230
 #5206

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5405)

### CODE REVIEW

**Comment by nebril** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Review by rlenglet — APPROVED** [review]

File: pkg/endpoint/bpf.go [hunk 1]
```diff
@@ -712,6 +712,20 @@ func (e *Endpoint) regenerateBPF(owner Owner, epdir, reason string) (uint64, boo
 		}
 	}
 
+	// Walk the L4Policy to add new redirects and update the desired policy map
```
Comment (rlenglet) [inline pkg/endpoint/bpf.go:715]: The changes to this file look suspicious. I would revert just those.

**Review by rlenglet — CHANGES_REQUESTED** [review]

File: pkg/endpoint/bpf.go (hunk 1 above)
Comment (jrfastab) [reply pkg/endpoint/bpf.go:715]: OK I'll drop the patch then.
https://github.com/cilium/cilium/pull/5405/commits/ea5ec53175d5783c44f90a50cc830b2bfdcaf1c4

Comment (rlenglet) [reply pkg/endpoint/bpf.go:715]: Yes, that's the one. Thanks!

**Review by rlenglet — APPROVED** [review]

**Comment by jrfastab** [comment]
New PR created here, https://github.com/cilium/cilium/pull/5443

**Comment by tgraf** [comment]
Closing in favor of #5443

**Pull request closed without merging**